build-formula.sh

```diff
@@ -132,22 +132,27 @@ echo "[${BOTTLE}]: Root URL -> ${BOTTLE_ASSET_URL}"
 BIN_NAME=$(echo ${BOTTLE_CONFIG} | jq -r '.bin')
 echo "[${BOTTLE}]: Bottle Binary -> ${BIN_NAME}"
 
-# Need to cd since brew bottle doesn't have an option to 
-cd "${BUILD_DIR}"
-brew bottle --no-rebuild --json --root-url="${BOTTLE_ASSET_URL}" "${TAP}/${BOTTLE}"
-cd ${SCRIPTPATH}
-
-RELEASE_FILE="$(ls ${BUILD_DIR}/${BOTTLE}--*.bottle.tar.gz)"
-# Renaming aws-sam-cli--0.37.0.sierra.bottle.tar.gz to aws-sam-cli-0.37.0.sierra.bottle.tar.gz
-rename 's/--/-/' ${BUILD_DIR}/*.bottle.* # replacing `--` with `-`
-RELEASE_FILE=$(echo ${RELEASE_FILE} | sed 's/--/-/')
-
-echo "[${BOTTLE}]: Brew bottles are built. Validating them now."
-echo "[${BOTTLE}]: Release file -> ${RELEASE_FILE}"
-
-brew uninstall -f ${BOTTLE}
-echo "[${BOTTLE}]: Installing release file"
-brew install ${RELEASE_FILE}
+if [[ $(grep -c 'bottle :unneeded' "${FORMULA_FILE}") -eq 0 ]]; then 
+  # Need to cd since brew bottle doesn't have an option to 
+  cd "${BUILD_DIR}"
+  brew bottle --no-rebuild --json --root-url="${BOTTLE_ASSET_URL}" "${TAP}/${BOTTLE}"
+  cd ${SCRIPTPATH}
+
+  RELEASE_FILE="$(ls ${BUILD_DIR}/${BOTTLE}--*.bottle.tar.gz)"
+  # Renaming aws-sam-cli--0.37.0.sierra.bottle.tar.gz to aws-sam-cli-0.37.0.sierra.bottle.tar.gz
+  rename 's/--/-/' ${BUILD_DIR}/*.bottle.* # replacing `--` with `-`
+  RELEASE_FILE=$(echo ${RELEASE_FILE} | sed 's/--/-/')
+
+  echo "[${BOTTLE}]: Brew bottles are built. Validating them now."
+  echo "[${BOTTLE}]: Release file -> ${RELEASE_FILE}"
+
+  brew uninstall -f ${BOTTLE}
+  echo "[${BOTTLE}]: Installing release file"
+  brew install ${RELEASE_FILE}
+else 
+  ## Touch a fake bottle so that build summary is accurate
+  touch "${BUILD_DIR}/${BOTTLE}.bottle.tar.gz"
```

Comment:
Why do we need to create an empty file here? Is this purely to make the build pass?

Reply:
yeah, not the cleanest solution I know. The build-bottles script will always continue formula builds even if they fail. Since `set -euo pipefail` is enabled, the build-formula command is executed with an ` build-formula <formula> || true` and then to check if it succeeds, it looks for the bottle file.  This seemed to be reasonable at the time when all formulas did output a bottle and there's a trap in the build-formula script to remove a bottle file if there is any failure in the build.  But now, maybe it makes sense to write a succeed file in the build dir instead of relying on the bottle to mean success.

Reply:
@jfuss updated to use a separate file to indicate success of indv builds instead of relying on the bottle output :)